### PR TITLE
Adding functional genomics

### PR DESCRIPTION
**Pull request by LaurenceKuhl**
This runs 3-4 modules related to functional genomics

### CODE REVIEW

File: nextflow_schema.json [hunk 1]
```diff
@@ -58,6 +42,62 @@
...
+                    "fa_icon": "fas fa-align-justify",
+                    "enum": ["minimap2", "bwa", "bowtie2"]
+                }
+            },
+            "required": ["aligner"]
+        },
+        "screening_parameters": {
+            "title": "Screening parameters",
+            "type": "object",
+            "description": "",
```
Comment (mirpedrol) [inline nextflow_schema.json:64]: could we add a description?

File: nextflow_schema.json [hunk 2]
```diff
@@ -58,6 +42,62 @@
...
+            "default": "",
+            "properties": {
+                "skip_alignment": {
+                    "type": "boolean",
```
Comment (mirpedrol) [inline nextflow_schema.json:68]: Description is also missing for most of the parameters

File: nextflow_schema.json [hunk 3]
```diff
@@ -58,6 +42,62 @@
...
+            },
+            "required": ["aligner"]
+        },
+        "screening_parameters": {
+            "title": "Screening parameters",
+            "type": "object",
+            "description": "",
+            "default": "",
+            "properties": {
+                "skip_alignment": {
```
Comment (mirpedrol) [inline nextflow_schema.json:67]: This parameter is missing a default value, `false` in `nextflow.config`

File: nextflow_schema.json [hunk 4]
```diff
@@ -58,6 +42,62 @@
...
+                },
+                "crisprcleanr": {
+                    "type": "string"
+                },
+                "min_reads": {
+                    "type": "number",
+                    "description": "a filter threshold value for sgRNAs, based on their average counts in the control sample",
+                    "default": 30
+                },
+                "min_targeted_genes": {
```
Comment (mirpedrol) [inline nextflow_schema.json:91]: default is missing, it's set to 3 in `nextflow.config`

**Review by mirpedrol — COMMENTED** [review]
It already looks very good! 🥳 
Most of my comments are regarding formatting.

As a reminder that came to my mind while reviewing, we will need to modify the GHA and `profile.test` to run tests with this workflow

File: lib/WorkflowMain.groovy [hunk 5]
```diff
@@ -80,10 +80,10 @@ class WorkflowMain {
         NfcoreTemplate.awsBatch(workflow, params)
 
         // Check input has been provided
-        if (!params.input) {
```
[the remark this answers is not in the document]
Comment (LaurenceKuhl) [reply lib/WorkflowMain.groovy:83]: Ok should I leave it like this in the meantime?

[the remark this answers is not in the document]
Comment (mirpedrol) [reply lib/WorkflowMain.groovy:83]: yes, but we should have it in mind :)

**Review by mirpedrol — APPROVED** [review]
LGTM if tests pass :)

**Pull request merged**